pkg/controller/node/controller_utils.go

```diff
@@ -243,6 +253,35 @@ func markAllPodsNotReady(kubeClient clientset.Interface, nodeName string) error
 	return fmt.Errorf("%v", strings.Join(errMsg, "; "))
 }
 
+//isPodForgivable will check whether Pod need to be forgive. Return true if pod has forgiveness toleration and duration is greater than forgivenessSeconds
+func isPodForgivable(pod api.Pod, nodeReadyTransitionTimestamp unversioned.Time) bool {
+	tolerations, err := api.GetTolerationsFromPodAnnotations(pod.ObjectMeta.Annotations)
+	if err != nil {
+		return false
+	}
+	var forgivenessToleration api.Toleration
+	found := false
+	for _, toleration := range tolerations {
+		if toleration.Key == unversioned.TaintNodeDown && toleration.Effect == api.TaintEffectNoExecute {
```

Comment:
@ingvagabund is it really necessary to list all possible taints in a switch? I was thinking about more generic approach:

1. find all taints that are not tolerated
2. find the lowest forgiveness period